packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -469,48 +489,170 @@ class HeapTreeViewState extends State<HeapTree>
         // tables objects/fields. Maybe notion in table?
         controller.showTreemap.value
             ? const SizedBox()
-            : OutlineButton(
-                key: collapseAllButtonKey,
-                onPressed: snapshotDisplay is MemorySnapshotTable
-                    ? () {
-                        if (snapshotDisplay is MemorySnapshotTable) {
-                          controller.groupByTreeTable.dataRoots
-                              .every((element) {
-                            element.collapseCascading();
-                            return true;
-                          });
-                          if (controller.instanceFieldsTreeTable != null) {
-                            // We're collapsing close the fields table.
-                            controller.selectedLeaf = null;
+            : Tooltip(
+                message: 'Collapse All',
+                child: OutlineButton(
+                  key: collapseAllButtonKey,
+                  onPressed: snapshotDisplay is MemorySnapshotTable
+                      ? () {
+                          if (snapshotDisplay is MemorySnapshotTable) {
+                            controller.groupByTreeTable.dataRoots
+                                .every((element) {
+                              element.collapseCascading();
+                              return true;
+                            });
+                            if (controller.instanceFieldsTreeTable != null) {
+                              // We're collapsing close the fields table.
+                              controller.selectedLeaf = null;
+                            }
+                            setState(() {});
```

Comment:
not new but while you are tweaking this code, can we remove the empty setState? or wrap the setState call around the state that is modified?

Reply:
Fixed the empty setState.